app/packages/looker/src/worker/label-3d-projection-utils.ts

```diff
@@ -61,23 +60,145 @@ export const getProjectedCorners = (
     // right top front
     [halfDimensions[0], halfDimensions[1], halfDimensions[2]],
   ];
+};
 
-  // rotate first, and translate
-  const transformedCorners = corners.map((corner) => {
-    const newRotation = rotation;
+export const translatePoint = (point: Vec3, translation: Vec3): Vec3 => {
+  return [
+    point[0] + translation[0],
+    point[1] + translation[1],
+    point[2] + translation[2],
+  ] as Vec3;
+};
 
+export const getGlobalPointsForCube = (
+  origin: Vec3,
+  corners: Vec3[],
+  rotation: Vec3
+): Vec3[] => {
+  return corners.map((corner) => {
+    const newRotation = rotation;
+    // rotate first
     const rotated = rotatePoint(corner, newRotation);
-    return [
-      rotated[0] + location[0],
-      rotated[1] + location[1],
-      rotated[2] + location[2],
-    ] as Vec3;
+    // then translate
+    return translatePoint(rotated, origin);
+  });
+};
+
+export const getCubeCorners = (box: BoundingBox3D) => {
+  const localCorners = getLocalBoundingBoxCorners3D(box);
+  const { location, rotation } = box;
+  return getGlobalPointsForCube(location, localCorners, rotation);
+};
+
+export const getProjectionPlaneForNormal = (normal: Vec3): ProjectionPlane => {
+  const [nx, ny, nz] = normal ?? [0, 0, 1];
```

Comment:
FYI- I added `?? [0, 0, 1]` here for backwards compatibility. `OrthographicProjectionMetadata` did not have a `normal` field in prior releases so we need to default here.